client/rest-high-level/src/main/java/org/elasticsearch/client/response/indices/flush/SyncedFlushResponse.java

```diff
@@ -0,0 +1,329 @@
+import java.util.ArrayList;
+
+public class SyncedFlushResponse extends ActionResponse implements ToXContentFragment {
+
+    // Field name declaration
+    public static final String SHARDS = "_shards";
+    // End field name declaration
+
+    private ShardCounts totalCounts;
+    private Map<String, IndexResult> indexResults;
+
+    SyncedFlushResponse(ShardCounts totalCounts, Map<String, IndexResult> indexResults) {
+        this.totalCounts = new ShardCounts(totalCounts.total, totalCounts.successful, totalCounts.failed);
+        this.indexResults = Collections.unmodifiableMap(indexResults);
+    }
+
+    public int totalShards() {
+        return totalCounts.total;
+    }
+
+    public int successfulShards() {
+        return totalCounts.successful;
+    }
+
+    public int failedShards() {
+        return totalCounts.failed;
+    }
+
+    public Map<String, IndexResult> getIndexResults() {
+        return indexResults;
+    }
+
+    protected ShardCounts getShardCounts() {
+        return totalCounts;
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject(SHARDS);
+        totalCounts.toXContent(builder, params);
+        builder.endObject();
+        for (Map.Entry<String, IndexResult> entry: indexResults.entrySet()) {
+            String indexName = entry.getKey();
+            IndexResult indexResult = entry.getValue();
+            builder.startObject(indexName);
+            indexResult.toXContent(builder, params);
+            builder.endObject();
+        }
+        return builder;
+    }
+
+    public static SyncedFlushResponse fromXContent(XContentParser parser) throws IOException {
+        ensureExpectedToken(Token.START_OBJECT, parser.nextToken(), parser::getTokenLocation);
+        ShardCounts totalCounts = null;
+        Map<String, IndexResult> indexResults = new HashMap<>();
+        XContentLocation startLoc = parser.getTokenLocation();
+        while (parser.nextToken().equals(Token.FIELD_NAME)) {
+            if (parser.currentName().equals(SHARDS)) {
+                ensureExpectedToken(Token.START_OBJECT, parser.nextToken(), parser::getTokenLocation);
+                totalCounts = ShardCounts.fromXContent(parser);
+                parser.nextToken();
+            } else {
+                String indexName = parser.currentName();
+                IndexResult indexResult = IndexResult.fromXContent(parser);
+                indexResults.put(indexName, indexResult);
+            }
+        }
+        if (totalCounts != null) {
+            return new SyncedFlushResponse(totalCounts, indexResults);
+        } else {
+            throw new ParsingException(
+                startLoc,
+                "Unable to reconstruct object. Total counts for shards couldn't be parsed."
+            );
+        }
+    }
+
+    private static int getIntValueFromField(XContentParser parser, String fieldName) throws IOException {
```

Comment:
I think `ObjectParser` or `ConstructingObjectParser` would probably be better here because they'll handle things like out order response.